packages/sources/cryptocompare-test/src/endpoint/crypto-router.ts

```diff
@@ -0,0 +1,21 @@
+import { wsTransport } from './crypto-ws'
+import { defaultEndpoint } from '../config'
+import { BatchEndpointTypes, cryptoInputParams } from '../crypto-utils'
+import { httpTransport } from './crypto'
+import { RoutingTransport } from '@chainlink/external-adapter-framework/transports/meta'
+import { CryptoPriceEndpoint } from '@chainlink/external-adapter-framework/adapter'
+
+const transports = {
+  ws: wsTransport,
+  rest: httpTransport,
+}
+
+export const routingTransport = new RoutingTransport<BatchEndpointTypes>(transports, {
+  defaultTransport: 'rest',
+})
+
+export const endpoint = new CryptoPriceEndpoint<BatchEndpointTypes>({
+  name: defaultEndpoint,
+  transport: routingTransport,
+  inputParameters: cryptoInputParams,
+})
```

Comment:
Since `WS_API_KEY` isn't required, we should add the customInputValidation to block requests to `ws` if the key isn't set here too like I mentioned in the `tradermade` PR

Reply:
Actually never mind. Just noticed the logic in the ws transport uses `API_KEY` as a fallback. You can disregard this.